rafthttp/http.go

```diff
@@ -135,16 +135,14 @@ func (h *streamHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	case path.Join(RaftStreamPrefix, string(streamTypeMessage)):
 		t = streamTypeMessage
 	default:
-		log.Printf("rafthttp: ignored unexpected streaming request path %s", r.URL.Path)
-		http.Error(w, "invalid path", http.StatusNotFound)
+		http.Error(w, "invalid path", http.StatusBadRequest)
```

Comment:
This checks the dir of the path, and it finds the dir doesn't match any known pattern. It cannot understand the path then, so it is bad request.